### PR TITLE
Guideline and workflow for internal developers

### PR DESCRIPTION
**Pull request by CunliangGeng**

### CODE REVIEW

**Comment by CunliangGeng** [comment]
@LilySnow you might be confused with `git clone`, `pip` and `conda`. 
`git clone` is to copy the Github online repo to your local machine, and then you can do changes of code in the local repo.
`pip` and `conda` are just tools to help install python packages. For DeepRank, we only release it for pypi but not conda, so we **must make sure `pip` work well to install dependencies**. But you could use `conda` during developing.

File: developer_guideline.md [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
-tutorial](https://help.github.com/articles/adding-an-existing-project-to-github-using-the-command-line/).
+To get onboard, first ask project manager to get access to the DeepRank repo (short for repository), then clone the repo to your local machine.
 
-Python versions
----------------
+**The development workflow is as the following**:
+1. Create an issue in [the repo Issues](https://github.com/DeepRank/deeprank/issues) for any idea, bug, feature, improvement, etc.
+2. Comment and mention (i.e. @) someone in the issue for discussion if necessary.
+3. Assign the issue to yourself or someone after asking
+4. Create a new branch from the `dev` branch for this issue, and name it in a format of `issue{ID}_*`, e.g. `issue7_format_docstring`.
```
Comment (NicoRenaud) [inline developer_guideline.md:22]: it's `developement` and  not `dev`

File: developer_guideline.md [hunk 2]
```diff
@@ -1,42 +1,36 @@
...
+5. Work in this issue branch to solve the issue. Check this [guide](https://google.github.io/eng-practices/review/developer/) to see how to make good commits.
+6. Pull request to merge this issue branch to the `dev` branch when last step is completed.
```
Comment (NicoRenaud) [inline developer_guideline.md:24]: **Create** a Pull Request

Comment (CunliangGeng) [reply developer_guideline.md]: Thanks, updated.

**Pull request merged**